File: 001-two-sum.py
#!/bin/python3

from typing import List
import math
import os
import random
import re
import sys

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        otherPart = dict()
        for i, num in enumerate(nums):
            second = target - num

            if second in otherPart:
                result.append(otherPart[second])
                result.append(i)
                break
            else:
                otherPart[num]=i
         

        return result
        

File: 001-two-sum.go
func twoSum(nums []int, target int) []int {
    other := make(map[int]int)
    
    for index, num := range nums {
        diff := target - num
        
        if otherIndex, ok := other[diff]; ok {
            return []int{otherIndex, index}
        }
        other[num] = index
    }
    return []int{}
}
